Realm/RLMResults.mm

```diff
@@ -563,6 +570,137 @@ - (RLMNotificationToken *)addNotificationBlock:(void (^)(RLMResults *, RLMCollec
     return _results.add_notification_callback(RLMWrapCollectionChangeCallback(block, self, true), std::move(keyPaths));
 }
 
+- (void)completeOnQueue:(dispatch_queue_t _Nullable)queue
```

Comment:
Taking a completion queue and passing the RLMResults to the completion callback is a bad combination. The RLMResults is only actually usable if it's queue-confined and the user explicitly passed in the same queue as they were already on. We should probably call the completion block on the Realm's scheduler rather than having a separate queue passed in. There probably also isn't really a reason to pass the RLMResults to the completion.

Reply:
So queue is still available from the user facing api? It's only the impl that's using the scheduler? So something like:
```
- (void)completeOnQueue:(dispatch_queue_t _Nullable)queue
             completion:(RLMResultsCompletionBlock)completion
                  error:(NSError *_Nullable)error {
    RLMResults *ret = (error != nil) ? nil : self;
    if (queue) {
        RLMScheduler* scheduler = [RLMScheduler dispatchQueue:queue];
        [scheduler invoke:^{
            completion(ret, error);
        }];
    } else {
        completion(ret, error);
    }
```

or this with the results removed

```
- (void)completeOnQueue:(dispatch_queue_t _Nullable)queue
             completion:(void(^)(NSError *))completion
                  error:(NSError *_Nullable)error {
    // the commented lines can also be removed right?, since [RLMScheduler dispatchQueue:] checks for queue?
    // if (queue) {
        RLMScheduler* scheduler = [RLMScheduler dispatchQueue:queue];
        [scheduler invoke:^{
            completion(error);
        }];
    // } else {
    //    completion(error);
    //}
}
```